cmd/gb-vendor/purge.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/constabulary/gb"
+	"github.com/constabulary/gb/cmd"
+	"github.com/constabulary/gb/cmd/gb-vendor/vendor"
+)
+
+func init() {
+	registerCommand(PurgeCmd)
+}
+
+func parseImports(root string) (map[string]struct{}, error) {
+	var found = struct{}{}            // Does not take any space
+	pkgs := make(map[string]struct{}) // Set
+
+	var walkFn = func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		if filepath.Ext(path) != ".go" { // Parse only go source files
+			return nil
+		}
+
+		fs := token.NewFileSet()
+		f, err := parser.ParseFile(fs, path, nil, parser.ImportsOnly)
+		if err != nil {
+			return err
+		}
+
+		for _, s := range f.Imports {
+			if _, ok := gb.Stdlib[s.Path.Value]; !ok {
+				pkgs[strings.Replace(s.Path.Value, "\"", "", -1)] = found
+			}
+		}
+		return nil
+	}
+
+	if err := filepath.Walk(root, walkFn); err != nil {
```

Comment:
If I `return pkgs, filepath.Walk(..)`, will the filepath.Walk(..) fill the `pkgs` correctly?